File: php/skill.php
#!/usr/bin/env php

<?php
$a = 1;
$b = 'foo';
$c = array('foo', 'bar', 'bazz');

#sample($a, $b, $c);
sample2($a, $b, $c);
#sample3($a, $b, $c);

echo "a is : $a", PHP_EOL;
echo "b is : $b", PHP_EOL;
echo "c is : " + var_dump($c), PHP_EOL;

function sample($a, $b, $c){
	 $a = 2;
	 $b = 'hoge';
	 $c = array('hoge', 'fuga', 'piyo');

	 echo "a is : $a", PHP_EOL;
	 echo "b is : $b", PHP_EOL;
	 echo "c is : "	 + var_dump($c), PHP_EOL;
	 
}

function sample2(&$a, &$b, &$c){
	 $a = 2;
	 $b = 'hoge';
	 array_push($c, array('hoge', 'fuga', 'piyo'));

	 echo "a is : $a", PHP_EOL;
	 echo "b is : $b", PHP_EOL;
	 echo "c is : "	 + var_dump($c), PHP_EOL;
	 
}

function sample3($a, $b, $c){
	 $a = 2;
	 $b = 'hoge';
	 array_push($c, array('hoge', 'fuga', 'piyo'));

	 echo "a is : $a", PHP_EOL;
	 echo "b is : $b", PHP_EOL;
	 echo "c is : "	 + var_dump($c), PHP_EOL;
	 
}

File: python/skill.py
#!/usr/bin/env python

a = 1
b = 'foo'
c = ['foo', 'bar', 'bazz']

def sample(a, b, c):
    a = 2
    b = 'hoge'
    c = ['hoge', 'fuga', 'buzz']
    
    print 'a is : ' + str(a)
    print 'b is : ' + b
    print 'c is : ' + str(c)

def sample2(a, b, c):
    a = 2
    b = 'hoge'
    c.extend( ['hoge', 'fuga', 'buzz'])
    
    print 'a is : ' + str(a)
    print 'b is : ' + b
    print 'c is : ' + str(c)

#sample(a, b, c)
sample2(a, b, c)

print 'a is : ' + str(a)
print 'b is : ' + b
print 'c is : ' + str(c)

    
